consonance/ml_logic/omr.py

```diff
@@ -4,35 +4,69 @@
 
 
 # Path to the data directory
-data_dir = Path("/Users/ninjamac/code/OMR-research/data")
+data_dir = Path("/Users/ninjamac/code/DataSets/images_cropped")
+
+# Load the CSV file
+csv_path = "/Users/ninjamac/code/DataSets/music/labels.csv"
+df = pd.read_csv(csv_path)
+
+
+# Extract filenames and labels
+images = [str(data_dir / filename) for filename in df['filename'].tolist()]
+
+# Convert the label strings into actual lists of integers
+labels = [eval(label) for label in df['labels'].tolist()]
 
-# Get list of all the images
-images = sorted(list(map(str, list(data_dir.glob("*.png")))))
-labels = [img.split(os.path.sep)[-1].split(".png")[0][:-2] for img in images]
-characters = set(char for label in labels for char in label)
+# Extract all unique label values (characters)
+characters = set()
+for label in labels:
+    characters.update(label)  # Add all elements of the label list to the set
+
+# Convert the set to a sorted list
 characters = sorted(list(characters))
 
+# Print out the results to verify
 print("Number of images found: ", len(images))
 print("Number of labels found: ", len(labels))
-print("Number of unique characters: ", len(characters))
-print("Characters present: ", characters)
+print("Number of unique characters (label values): ", len(characters))
+print("Unique label values (characters): ", characters)
+
+
+# Feed in label dictionary
+combined_dict = {
+    'B3_whole': 0, 'C4_whole': 1, 'D4_whole': 2, 'E4_whole': 3, 'F4_whole': 4, 'G4_whole': 5, 'A4_whole': 6, 'B4_whole': 7, 'C5_whole': 8,
+    'D5_whole': 9, 'E5_whole': 10, 'F5_whole': 11, 'G5_whole': 12, 'A5_whole': 13, 'B5_whole': 14, 'C6_whole': 15, 'D6_whole': 16,
+
+    'B3_half': 17, 'C4_half': 18, 'D4_half': 19, 'E4_half':20, 'F4_half': 21, 'G4_half': 22, 'A4_half': 23, 'B4_half': 24, 'C5_half': 25,
+    'D5_half': 26, 'E5_half': 27, 'F5_half': 28, 'G5_half': 29, 'A5_half': 30, 'B5_half': 31, 'C6_half': 32, 'D6_half': 33,
+
+    'B3_quarter': 34, 'C4_quarter': 35, 'D4_quarter': 36, 'E4_quarter': 37, 'F4_quarter': 38, 'G4_quarter': 39, 'A4_quarter': 40, 'B4_quarter': 41, 'C5_quarter': 42,
+    'D5_quarter': 43, 'E5_quarter': 44, 'F5_quarter': 45, 'G5_quarter': 46, 'A5_quarter': 47, 'B5_quarter': 48, 'C6_quarter': 49, 'D6_quarter': 50,
+
+    'B3_eighth': 51, 'C4_eighth': 52, 'D4_eighth': 53, 'E4_eighth': 54, 'F4_eighth': 55, 'G4_eighth': 56, 'A4_eighth': 57, 'B4_eighth': 58, 'C5_eighth': 59,
+    'D5_eighth': 60, 'E5_eighth': 61, 'F5_eighth': 62, 'G5_eighth': 63, 'A5_eighth': 64, 'B5_eighth': 65, 'C6_eighth': 66, 'D6_eighth': 67,
+
+    'B3_16th': 68, 'C4_16th': 69, 'D4_16th': 70, 'E4_16th': 71, 'F4_16th': 72, 'G4_16th': 73, 'A4_16th': 74, 'B4_16th': 75, 'C5_16th': 76,
+    'D5_16th': 77, 'E5_16th': 78, 'F5_16th': 79, 'G5_16th': 80, 'A5_16th': 81, 'B5_16th': 82, 'C6_16th': 83, 'D6_16th': 84
+}
 
 # Batch size for training and validation
 batch_size = 16
 
 # Desired image dimensions
-img_width = 500
+img_width = 350
```

Comment:
I assume I need to change this value in preprocessing for the prediction too?

Reply:
Any prediction would need to follow the dimensions that the model was trained on, yes. This would be part of preprocessing for any image we run a prediction on.